display/ssd1306.h

```diff
@@ -274,10 +305,27 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
   }
 
   void SB_NewFont() override {
-    if (IMG_font) {
-      // Overrides message from below..
-      ShowFile(&IMG_font);
-    }
+    ShowFile(&IMG_font);
+  }
+
+  void SB_On() override {
+    ShowFile(&IMG_on);
+  }
+
+  void SB_Blast() override {
+    ShowFile(&IMG_blst);
+  }
+
+  void SB_Clash() override {
+    ShowFile(&IMG_clsh);
+  }
+
+  void SB_BeginLockup() override {
+    ShowFile(&IMG_lock);
+  }
+
+  void SB_EndLockup() override {
+    ShowFile(&IMG_on);
```

Comment:
It seems to me that the "on" image is not looping, then we only want to play it once when turning on.

Reply:
I'm struggling to come up with a good way to do a check to see if a on.bmp is static when looking to play the on.bmp after an effect (blast, clash, lockup)  Any tips on how you'd do it?

Reply:
I'm thinking we need another bool. (Maybe called on_is_looped_)
When we calculate if the current frame is looped or not, we then do:

   if (is_on_) { on_is_looped_ = looped_frames_ > 1; }

We would need to reset on_is_looped_ in SB_NewFont()

Reply:
Of course, this makes the assumption that a looping ON image means that we need to resume it after lockup, and that a non-looping ON image means that we don't.  Not entirely sure if that's the assumption we should make though...

Reply:
My working assumption was that looping on.bmp would always play, except when interrupted by blst, clsh or lock, and then resume. We'd only want the static on.bmp to play once and revert to PLI.

What do you think would be the right approach here?

Reply:
That's basically my assumption too. I just wanted to check my assumptions. :)